src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp

```diff
@@ -262,12 +262,24 @@ namespace OpenMS
       // and all bytes except the least significant one will be zero. Thus
       // we can convert to char directly (only keeping the least
       // significant byte).
+      const XMLCh* it = chars;
+      const XMLCh* end = it + length;
+
       size_t curr_size = result.size();
       result.resize(curr_size + length);
-      for (size_t i = 0; i < length; i++)
-      {
-        result[curr_size + i] = (char)chars[i];
+      std::string::iterator str_it = result.begin();
+      std::advance(str_it, curr_size);
+      while (it!=end)
+      {   
+        *str_it++ = (char)*it++;
```

Comment:
it is about the same but it may be more intuitive and easier to read (see http://www.cplusplus.com/reference/algorithm/copy/ )

`*str_it++ = (char)*it++;`:
```
real    1m36.645s
user    1m20.483s
sys     0m15.997s

real    1m32.173s
user    1m17.925s
sys     0m14.107s

real    1m38.621s
user    1m20.615s
sys     0m17.864s
```

vs
`*str_it = (char)*it; ++str_it;++it;`

```
real    1m30.579s
user    1m16.497s
sys     0m13.887s

real    1m35.613s
user    1m19.135s
sys     0m16.326s

real    1m36.674s
user    1m19.245s
sys     0m17.292s

```